reference/test/matrix/dense_kernels.cpp

```diff
@@ -4251,4 +4266,22 @@ TYPED_TEST(DenseComplex, ConjDot)
 }
 
 
+TEST(DenseComplex, ScaleAddIdentityRectangularMixed)
```

Comment:
maybe not neccessary because Scalar and ValueType will be the same

Reply:
What do you mean? In the test, the scalar type is `complex<double>` but the matrix value type is `complex<float>`. I guess this test is to make sure that if the user does pass a different scalar type, it compiles and works.

I agree that the extra template parameter on the kernels is useless right now, but having the extra generality would not hurt, I guess; at some point, we might need to enable ScalarType != ValueType.

Reply:
I read the `GKO_INSTANTIATE_FOR_EACH_VALUE_AND_SCALAR_TYPE` again and realize my assumption is wrong.
`GKO_INSTANTIATE_FOR_EACH_VALUE_AND_SCALAR_TYPE` only generates additional type pair for real scalar but complex data. not for all possible precision combination.
By the way, are precision_dispatch_real_complex a and b always the same type?